tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/LocaleCompletionCounter.java

```diff
@@ -1,119 +1,49 @@
 package org.unicode.cldr.web.api;
 
 import com.ibm.icu.dev.util.ElapsedTimer;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.EnumSet;
 import java.util.logging.Logger;
-import org.unicode.cldr.test.CheckCLDR;
-import org.unicode.cldr.test.CoverageLevel2;
-import org.unicode.cldr.test.SubmissionLocales;
-import org.unicode.cldr.test.TestCache;
 import org.unicode.cldr.util.*;
-import org.unicode.cldr.web.STFactory;
-import org.unicode.cldr.web.SurveyLog;
-import org.unicode.cldr.web.SurveyMain;
+import org.unicode.cldr.web.*;
 
 public class LocaleCompletionCounter {
 
     private static final Logger logger = SurveyLog.forClass(LocaleCompletionCounter.class);
 
+    private static final EnumSet<VettingViewer.Choice> choiceSet = EnumSet.of(
+        VettingViewer.Choice.error,
+        VettingViewer.Choice.hasDispute,
+        VettingViewer.Choice.notApproved,
+        VettingViewer.Choice.missingCoverage
+    );
+
     private final String localeId;
     private final Level level;
-    private final CheckCLDR.Options options;
-    private final TestCache.TestResultBundle checkCldr;
-    private final CLDRFile file;
-    private final CLDRFile baselineFile;
     private final LocaleCompletion.LocaleCompletionResponse lcr;
-    private final CoverageLevel2 covLeveller;
-    private final List<CheckCLDR.CheckStatus> results;
+    private final VettingViewer<Organization> vv;
+    private final VettingViewer.DashboardArgs args;
 
     public LocaleCompletionCounter(CLDRLocale cldrLocale, STFactory stFactory) {
-        localeId = cldrLocale.toString(); // normalized
+        localeId = cldrLocale.toString();
         level = StandardCodes.make().getTargetCoverageLevel(localeId);
-        options = new CheckCLDR.Options(cldrLocale, SurveyMain.getTestPhase(), level.toString(), null);
-        checkCldr = stFactory.getTestResult(cldrLocale, options);
-
-        // we need an XML Source to receive notification.
-        // This causes LocaleCompletionHelper.INSTANCE.valueChanged(...) to be called
-        // whenever a vote happens.
-        final XMLSource mySource = stFactory.makeSource(localeId, false);
-        mySource.addListener(LocaleCompletion.LocaleCompletionHelper.INSTANCE);
-
-        file = stFactory.make(localeId, true);
-        baselineFile = stFactory.getDiskFile(cldrLocale);
         lcr = new LocaleCompletion.LocaleCompletionResponse(level);
-        SupplementalDataInfo sdi = SupplementalDataInfo.getInstance(stFactory.getSupplementalDirectory());
-        covLeveller = CoverageLevel2.getInstance(sdi, localeId);
-        results = new ArrayList<>();
+        final SurveyMain sm = CookieSession.sm;
+        vv = new VettingViewer<>(sm.getSupplementalDataInfo(), stFactory, new STUsersChoice(sm));
+        args = new VettingViewer.DashboardArgs(choiceSet, cldrLocale, level);
+        args.setUserAndOrganization(0, VettingViewer.getNeutralOrgForSummary());
+        Dashboard.setFiles(args, cldrLocale, stFactory);
```

Comment:
@btangmu  tracing this call through more…  wouldn't this function be better as `args.setFiles(cldrLocale, stFactory)`?  ( make it a member function on DashboardArgs not a static on Dashboard)